lib/axiom/fs/stream/writable_file_stream_buffer.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AxiomError from 'axiom/core/error';
+import AxiomEvent from 'axiom/core/event';
+
+import DataType from 'axiom/fs/data_type';
+import SeekWhence from 'axiom/fs/seek_whence';
+import StreamBuffer from 'axiom/fs/stream/stream_buffer';
+import WritableMemoryStream from 'axiom/fs/stream/writable_memory_stream';
+
+/** @typedef FileSystem$$module$axiom$fs$base$file_system */
+var FileSystem;
+
+/** @typedef OpenMode$$module$axiom$fs$open_mode */
+var OpenMode;
+
+/** @typedef Path$$module$axiom$fs$path */
+var Path;
+
+/** @typedef WritableStream$$module$axiom$fs$stream$writable_stream */
+var WritableStream;
+
+/**
+ * @constructor @extends {StreamBuffer}
+ * @param {!ExecuteContext} execCx
+ * @param {Path} filePath
+ * @param {!OpenMode} fileMode
+ */
+export var WritableFileStreamBuffer = function(fileSystem, filePath, fileMode) {
+  StreamBuffer.call(this);
+
+  /** @const @type {!WritableStream} */
+  this.writableStream = new WritableMemoryStream(this);
+
+  this.onClose = new AxiomEvent();
+
+  fileSystem.createOpenContext(filePath, fileMode)
```

Comment:
Same comment as `ReadableFileStreamBuffer`: I don't think we need this class, we can just have a `createWritableStream` on `FileSystem` with a default implementation.